yt/yt/server/node/exec_node/job.cpp

```diff
@@ -133,6 +133,8 @@ using NCypressClient::EObjectType;
 
 static constexpr auto DisableSandboxCleanupEnv = "YT_DISABLE_SANDBOX_CLEANUP";
 
+static const TString ShardingKeyPattern("\%sharding_key\%");
```

Comment:
I think this should be part of the `SeparateDirectory` logging mode. I mean that we should provide user to configure some tricky schemes with `%sharding_key%` somewhere in the path